h5netcdf/tests/test_h5netcdf.py

```diff
@@ -571,3 +574,167 @@ def test_invalid_then_valid_no_ncproperties(tmp_netcdf):
+        f.dimensions['z'] = None
+        f.resize_dimension('z', 20)
+
+        with pytest.raises(ValueError) as e:
+            f.resize_dimension('y', 20)
+        assert e.value.args[0] == (
+            "Dimension 'y' is not unlimited and thus cannot be resized.")
+
+    # Assert some behavior observed by using the C netCDF bindings.
+    with h5py.File(tmp_netcdf) as f:
+        assert f["x"].shape == (0,)
+        assert f["x"].maxshape == (None,)
+        assert f["y"].shape == (15,)
+        assert f["y"].maxshape == (15,)
+        assert f["z"].shape == (20,)
+        assert f["z"].maxshape == (None,)
+
+def test_creating_variables_with_unlimited_dimensions(tmp_netcdf):
+    with h5netcdf.File(tmp_netcdf) as f:
+        f.dimensions['x'] = None
+        f.dimensions['y'] = 2
+
+        # Creating a variable without data will initialize an array with zero
+        # length.
+        f.create_variable('dummy', dimensions=('x', 'y'), dtype=np.int64)
+        assert f.variables["dummy"].shape == (0, 2)
+        assert f.variables["dummy"]._h5ds.maxshape == (None, 2)
+
+        # Trying to create a variable while the current size of the dimension
+        # is still zero will fail.
+        with pytest.raises(ValueError) as e:
+            f.create_variable('dummy2', data=np.array([[1, 2], [3, 4]]),
+                              dimensions=('x', 'y'))
+        assert e.value.args[0] == "Shape tuple is incompatible with data"
+
+        # Resize data.
+        assert f.variables["dummy"].shape == (0, 2)
+        f.resize_dimension('x', 3)
+        # This will also force a resize of the existing variables and it will
+        # be padded with zeros..
+        np.testing.assert_allclose(f.variables["dummy"], np.zeros((3, 2)))
+
+        # Creating another variable with no data will now also take the shape
+        # of the current dimensions.
+        f.create_variable('dummy3', dimensions=('x', 'y'), dtype=np.int64)
+        assert f.variables["dummy3"].shape == (3, 2)
+        assert f.variables["dummy3"]._h5ds.maxshape == (None, 2)
+
+    # Close and read again to also test correct parsing of unlimited
+    # dimensions.
+    with h5netcdf.File(tmp_netcdf) as f:
+        assert f.dimensions['x'] is None
+        assert f._h5file['x'].maxshape == (None,)
+        assert f._h5file['x'].shape == (3,)
+
+        assert f.dimensions['y'] == 2
+        assert f._h5file['y'].maxshape == (2,)
+        assert f._h5file['y'].shape == (2,)
+
+def test_writing_to_an_unlimited_dimension(tmp_netcdf):
+    with h5netcdf.File(tmp_netcdf) as f:
+        # Two dimensions, only one is unlimited.
+        f.dimensions['x'] = None
+        f.dimensions['y'] = 3
+
+        # Cannot create it without first resizing it.
+        with pytest.raises(ValueError) as e:
+                f.create_variable('dummy1', data=np.array([[1, 2, 3]]),
+                                  dimensions=('x', 'y'))
+                assert e.value.args[0] == \
```

Comment:
Use parentheses for line continuation instead of explicit `\`:
https://www.python.org/dev/peps/pep-0008/#maximum-line-length

> The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.